Automattic/woocommerce-payments
# Use date picker for applicable dispute evidence fields

Fixes #505

#### Changes proposed in this Pull Request

Used native date input fields instead of text ones for shipping date and service date fields in the "Challenge Dispute" section. This allows a user to select the date from the date picker instead of typing.

I've chosen native date fields over the `DatePicker` component in `@woocommerce/components` due to the complexity of component setup and the need to write additional code in the widget.

The date range for date input is restricted by today, so we can not choose the date in the future. But there is a hack, a user can type the date in the future manually.

Small change: removed a link to the outdated Open a PR guide in the PR template.

#### Screenshots 
Service date
![screenshot](https://d.pr/i/vMIC0u+)

Shipping date
![screenshot](https://d.pr/i/aAfR7U+)

![screenshot](https://d.pr/i/23mMUR+)
![screenshot](https://d.pr/i/Sy7b46+)

#### Testing instructions





* Select `Product Type`: `Physical product` on the "Challenge dispute" page. Select the value for the `Date of shipment` field in the `Shipping information` field. Click the `Save for later` button and check if it is saved.
* Select `Product Type`: `Offline product` on the "Challenge dispute" page. Select the value for the `Service date` field in the `Service details` section. Click the `Save for later` button and check if it is saved.
* Select `Product Type`: `Multiple product types` on the "Challenge dispute" page. Select the value for the `Service date` field in the `Service details` section. Select the value for the `Date of shipment` field in the `Shipping information` field. Click the `Save for later` button and check if both values are saved.
* Test that you can not choose the date in the future with any date input. The future dates must be disabled.
-------------------
